packages/forms/src/UIForm/fields/Datalist/Datalist.component.js

```diff
@@ -74,25 +77,50 @@ class Datalist extends Component {
 	}
 
 	getTitleMap() {
-		const titleMap = this.state.titleMap || this.props.schema.titleMap || [];
+		const isMultiSection = this.props.schema.options && this.props.schema.options.multiSection;
+		const titleMap =
+			this.state.titleMap ||
+			((this.props.schema.options && this.props.schema.options.titleMap) ||
+				this.props.schema.titleMap) ||
+			[];
+		let titleMapFind = titleMap;
 
 		if (!this.props.schema.restricted) {
 			const isMultiple = this.props.schema.schema.type === 'array';
 			const values = isMultiple ? this.props.value : [this.props.value];
+
+			if (isMultiSection) {
+				titleMapFind =
+					titleMap &&
+					titleMap.reduce((prev, current) => {
+						prev.push(...current.suggestions);
+						return prev;
+					}, []);
+			}
+
 			const additionalOptions = values
 				.filter(value => value)
-				.filter(value => !titleMap.find(option => option.value === value))
-				.map(value => ({ name: this.props.resolveName(value), value }))
+				.filter(value => !titleMapFind.find(option => option.value === value))
+				.map(value => this.addCustomValue(value, isMultiSection))
 				.reduce((acc, titleMapEntry) => {
 					acc.push(titleMapEntry);
 					return acc;
 				}, []);
 			return titleMap.concat(additionalOptions);
 		}
-
 		return titleMap;
 	}
 
+	addCustomValue(value, multiSection) {
```

Comment:
i think the `multisection` parameter should be renamed `isMultisection` for consistency sake.